networkx/algorithms/core.py

```diff
@@ -460,25 +532,30 @@ def onion_layers(G):
     Parameters
     ----------
     G : NetworkX graph
-        A simple graph without self loops or parallel edges
+        An undirected graph without self loops.
 
     Returns
     -------
     od_layers : dictionary
-        A dictionary keyed by vertex to the onion layer. The layers are
+        A dictionary keyed by node to the onion layer. The layers are
         contiguous integers starting at 1.
 
     Raises
     ------
     NetworkXError
-        The onion decomposition is not implemented for graphs with self loops
-        or parallel edges or for directed graphs.
+        The onion decomposition is not implemented for graphs with self loops.
+    NetworkXNotImplemented
+        If `G` is a Multigraph or Directed graph.
```

Comment:
Let's make both of these errors `NetworkXNotImplemented` (line 581).
The other functions can be handled similarly.  The history is that `NetworkXError` used to be the only exception type with messages determining what the problem is. We shifted to multiple exception types later and have only incrementally been updating exceptions. This seems like a good time to do that.